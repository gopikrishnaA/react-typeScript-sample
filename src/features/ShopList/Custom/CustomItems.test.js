import React from "react";
import { shallow } from "enzyme";
import renderer from 'react-test-renderer';
import CustomItems from "./CustomItems";

describe("CustomItems", () => {
  let wrapper;

  let props = {
    items: [{
      amount: 1000,
      unit: 1,
      name: 'test'
    }, {
      amount: 10000,
      unit: 10,
      name: 'test10'
    }],
    removeCustomItem: jest.fn()
  };

  beforeEach(() => {
    wrapper = shallow(<CustomItems {...props} />);
  });

  describe("on start", () => {
    it("renders correctly", () => {
      const tree = renderer
        .create(wrapper.find(".custom-list"))
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
    it("should click on  particular ingredient", () => {
      wrapper.find(".fa-trash").at(0).simulate('click');
      expect(props.removeCustomItem).toHaveBeenCalledWith(0);
    })
  });
});
