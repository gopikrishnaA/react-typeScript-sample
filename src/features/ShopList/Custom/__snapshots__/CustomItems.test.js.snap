// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomItems on start renders correctly 1`] = `
<div
  className="mt-3 custom-list"
>
  <h6>
    Additionally: 
  </h6>
  <div
    className="shop-item p-1 mb-2 d-flex align-items-center justify-content-between"
  >
    1000
     
    <span
      className="ml-1 flex-1"
    >
      1
       
      test
    </span>
    <i
      className="shop-item__remove fas fa-trash"
      onClick={[Function]}
    />
  </div>
  <div
    className="shop-item p-1 mb-2 d-flex align-items-center justify-content-between"
  >
    10000
     
    <span
      className="ml-1 flex-1"
    >
      10
       
      test10
    </span>
    <i
      className="shop-item__remove fas fa-trash"
      onClick={[Function]}
    />
  </div>
</div>
`;
